crates/runtime/tests/iceberg/mod.rs

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2024-2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use crate::{get_test_datafusion, init_tracing, utils::test_request_context};
+use app::AppBuilder;
+use arrow::array::RecordBatch;
+use datafusion::assert_batches_eq;
+use futures::StreamExt;
+use runtime::extension::ExtensionFactory;
+use runtime::{status, Runtime};
+use spice_cloud::SpiceExtensionFactory;
+use spicepod::component::{catalog::Catalog, dataset::Dataset, params::Params as DatasetParams};
+use std::collections::HashMap;
+use std::sync::Arc;
+
+#[tokio::test]
+async fn iceberg_integration_test_dataset() -> Result<(), anyhow::Error> {
+    let _ = rustls::crypto::CryptoProvider::install_default(
+        rustls::crypto::aws_lc_rs::default_provider(),
+    );
+    let _tracing = init_tracing(None);
+
+    let account_id = std::env::var("AWS_ICEBERG_ACCOUNT_ID")
+        .map_err(|_| anyhow::anyhow!("AWS_ICEBERG_ACCOUNT_ID is not set"))?;
+    let region = std::env::var("AWS_ICEBERG_REGION")
+        .map_err(|_| anyhow::anyhow!("AWS_ICEBERG_REGION is not set"))?;
+    let access_key_id = std::env::var("AWS_ACCESS_KEY_ID")
+        .map_err(|_| anyhow::anyhow!("AWS_ACCESS_KEY_ID is not set"))?;
+    let secret_access_key = std::env::var("AWS_SECRET_ACCESS_KEY")
+        .map_err(|_| anyhow::anyhow!("AWS_SECRET_ACCESS_KEY is not set"))?;
```

Comment:
```suggestion
    let account_id = std::env::var("AWS_ICEBERG_ACCOUNT_ID")
        .context("AWS_ICEBERG_ACCOUNT_ID is not set")?;
    let region = std::env::var("AWS_ICEBERG_REGION")
        .context("AWS_ICEBERG_REGION is not set")?;
    let access_key_id = std::env::var("AWS_ACCESS_KEY_ID")
        .context("AWS_ACCESS_KEY_ID is not set")?;
    let secret_access_key = std::env::var("AWS_SECRET_ACCESS_KEY")
        .context("AWS_SECRET_ACCESS_KEY is not set")?;
```

nit: `anyhow::Context` is equivalent?